### PR TITLE
Hl 1069 ahjo cancel

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
Adjusts the callback view so that different kinds of callbacks can be received based on the AhjoRequestType enum.
Add a callback address for a DELETE request to Ahjo REST api.

Refactor the function `do_ahjo_request_with_json_payload` into `send_request_to_ahjo` that can do different types of requests.

## Testing :alembic:
```
pytest applications/tests/test_ahjo_integration.py
pytest applications/tests/test_ahjo_requests.py

```
## Additional notes :spiral_notepad:
`ahjo_integration_views.py` has a function, `_handle_delete_callback()` which currently does nothing, as placeholder because it's not certain yet, what we should do to the application after it is deleted in Ahjo.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2589.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2589.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2589.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2589.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2589.dev.hel.ninja :laughing::tada::tada::tada:

File: backend/benefit/applications/api/v1/ahjo_integration_views.py [hunk 1]
```diff
@@ -97,21 +108,28 @@ def post(self, request, *args, **kwargs):
...
+                application = self._handle_open_case_callback(
+                    application, callback_data
+                )
+                ahjo_status = AhjoStatusEnum.CASE_OPENED
+                info = f"""Application ahjo_case_guid and ahjo_case_id
+                were updated by Ahjo request id: {ahjo_request_id}"""
+            elif request_type == AhjoRequestType.DELETE_APPLICATION:
+                self._handle_delete_callback()
+                ahjo_status = AhjoStatusEnum.DELETE_REQUEST_RECEIVED
+                info = f"""Application was marked for cancellation in Ahjo with request id: {ahjo_request_id}"""
```
Comment (mjturt) [inline backend/benefit/applications/api/v1/ahjo_integration_views.py:123]: Why triple `"`'s?

**Review by mjturt — APPROVED** [review]

**Pull request merged**